src/lib/operators/join_hash.cpp

```diff
@@ -244,8 +244,8 @@ class JoinHash::JoinHashImpl : public AbstractJoinOperatorImpl {
   }
 
   template <typename T>
-  RadixContainer<T> _partition_radix_parallel(std::shared_ptr<Partition<T>> materialized,
-                                              std::shared_ptr<std::vector<size_t>> chunk_offsets,
+  RadixContainer<T> _partition_radix_parallel(const std::shared_ptr<Partition<T>>& materialized,
```

Comment:
If `materialized` becomes a `const std::shared_ptr&`, it will be a reference in the lambda defined for the JobTask in line 292. If that lambda throws an exception, the exception will kill the JoinHash, freeing the partition. Other threads that still run will now read from freed memory.